components/eventing-controller/controllers/subscription/eventmesh/reconciler_internal_integration_test.go

```diff
@@ -175,6 +176,288 @@ func TestReconciler_Reconcile(t *testing.T) {
+				te.credentials = credentials
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("SyncSubscription", mock.Anything, mock.Anything, mock.Anything).Return(true, nil)
+				return NewReconciler(ctx, te.fakeClient, te.logger, te.recorder, te.cfg, te.cleaner, te.backend,
+					te.credentials, te.mapper, validator), te.fakeClient
+			},
+			givenEventingWebhookAuthEnabled: false,
+			wantReconcileResult:             ctrl.Result{},
+			wantReconcileError:              nil,
+			wantHandler: apigatewayv1beta1.Handler{
+				Name:   object.OAuthHandlerNameOAuth2Introspection,
+				Config: nil,
+			},
+		},
+		{
+			name:              "Eventing webhook auth is enabled",
+			givenSubscription: subscription,
+			givenReconcilerSetup: func() (*Reconciler, client.Client) {
+				te := setupTestEnvironment(t, subscription)
+				te.credentials = credentials
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("SyncSubscription", mock.Anything, mock.Anything, mock.Anything).Return(true, nil)
+				return NewReconciler(ctx, te.fakeClient, te.logger, te.recorder, te.cfg, te.cleaner, te.backend,
+					te.credentials, te.mapper, validator), te.fakeClient
+			},
+			givenEventingWebhookAuthEnabled: true,
+			wantReconcileResult:             ctrl.Result{},
+			wantReconcileError:              nil,
+			wantHandler: apigatewayv1beta1.Handler{
+				Name: object.OAuthHandlerNameJWT,
+				Config: &runtime.RawExtension{
+					Raw: []byte(fmt.Sprintf(object.JWKSURLFormat, credentials.CertsURL)),
+				},
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		tc := testCase
+		t.Run(tc.name, func(t *testing.T) {
+			// given
+			featureflags.SetEventingWebhookAuthEnabled(tc.givenEventingWebhookAuthEnabled)
+			reconciler, cli := tc.givenReconcilerSetup()
+			namespacedName := k8stypes.NamespacedName{
+				Namespace: tc.givenSubscription.Namespace,
+				Name:      tc.givenSubscription.Name,
+			}
+
+			// when
+			res, err := reconciler.Reconcile(context.Background(), ctrl.Request{NamespacedName: namespacedName})
+			require.Equal(t, tc.wantReconcileResult, res)
+			require.Equal(t, tc.wantReconcileError, err)
+
+			sub := &eventingv1alpha2.Subscription{}
+			err = cli.Get(ctx, namespacedName, sub)
+			require.NoError(t, err)
+
+			namespacedName = k8stypes.NamespacedName{
+				Namespace: sub.Namespace,
+				Name:      sub.Status.Backend.APIRuleName,
+			}
+
+			apiRule := &apigatewayv1beta1.APIRule{}
+			err = cli.Get(ctx, namespacedName, apiRule)
+			require.NoError(t, err)
+
+			// then
+			require.Equal(t, tc.wantHandler.Name, apiRule.Spec.Rules[0].AccessStrategies[0].Handler.Name)
+			require.Equal(t, tc.wantHandler.Config, apiRule.Spec.Rules[0].AccessStrategies[0].Handler.Config)
+		})
+	}
+}
+
+// TestReconciler_APIRuleConfig_Upgrade ensures that the created APIRule is configured correctly
+// before and after the upgrade from ory to Eventing webhook auth and vise versa.
```

Comment:
I thought we will not downgrade to ory. Is it a valid use case?

Reply:
As long as we have a feature flag, we must test switching it on and off. In this case, it means deciding to use which secret.

Reply:
as far as I understand, the goal of the feature flag is to not run your new code until other side is ready. If other side were ready, you'd we'd always enable feature flag as ORY will be deleted soon. Hence, I don't see a real use case for going back. 
Although having the test case is little bit more code, but  it is minor thing. Hence, you can keep the test case too.

Reply:
In genera l agreed with your comment, but also the edge cases should be tested, and IMO switching the feature flag on/off (back and forth) should be tested.